theano/gpuarray/subtensor.py

```diff
@@ -582,7 +584,14 @@ class GpuAdvancedIncSubtensor1(Op):
     """
     _f16_ok = True
     __props__ = ('inplace', 'set_instead_of_inc')
-    params_type = gpu_context_type
+    params_type = ParamsType(inplace=bool_t,
+                             set_instead_of_inc=bool_t,
+                             context=gpu_context_type,
+                             # following params are for c_init_code_struct.
+                             ndim_input_0=size_t,
+                             ndim_input_1=size_t,
+                             typecode_input_0=int_t,
+                             typecode_input_1=int_t)
```

Comment:
There are some C codes when runtime inputs are not available but we have access to params. For example, in c_init_code_struct() ( https://github.com/Theano/Theano/blob/master/theano/gpuarray/subtensor.py#L361 ). So, in such cases, using params may help avoid hardcoding input properties.

But I will be more strict. Indeed, the general rule should be that params are Op props + gpu_context for gpuarray ( + handle_type for cuDNN ), and should not bundle any property that is reachable at runtime.